packages/babel-parser/src/parser/statement.ts

```diff
@@ -914,12 +914,11 @@ export default abstract class StatementParser extends ExpressionParser {
 
     const startsWithLet = this.isContextual(tt._let);
     const startsWithUsing =
-      this.isContextual(tt._using) && !this.hasFollowingLineBreak();
+      this.isContextual(tt._using) &&
+      this.hasInLineFollowingBindingIdentifier();
```

Comment:
Can we move this check again to after `startsWithUsing` in line 921 below? We don't need to run it when `isLetOrUsing` is true because of `startsWithLet`.

Reply:
Actually the line break check here is not required.
```js
for (const
  x of []);
```
is valid, and so should
```js
// Currently throwing "Missing semicolon. (1:10)"
for (using
  x of []);
```

Reply:
The proposal spec requires it; @rbuckton is there a reason?

Reply:
Ah you are right. I overlooked the spec.

Line break is probably safe here. `for (using` requires a trailing semicolon to start a ForStatement, so there won't be ASI issues here from `using` not being a keyword.

As for this PR I will keep the current behaviour.